### PR TITLE
Refactor and add tests for About You summary view

### PR DESCRIPTION
**Pull request by paulodeon**
The view components have gotten unwieldy and were untested, this PR starts to tidy them up by adding some view helpers and tests for the view components

https://trello.com/c/CIr7zEGo/1337-epic-view-component-refactor

### CODE REVIEW

File: spec/components/about_you_component_spec.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    it "renders the phone row" do
+      expect(page).to have_css("dd", text: referrer.phone)
+      expect(page).to have_link "Change", href: "/referrals/#{referral.id}/referrer/phone/edit?return_to=%2F"
+    end
+  end
+
+  context "with a public referral" do
+    let(:referral) { create(:referral, :public_complete) }
+
+    it "doesnt render the job title row" do
```
Comment (tavy88) [inline spec/components/about_you_component_spec.rb:37]: typo here

File: spec/components/about_you_component_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  before { render_inline(component) }
+
+  context "with an employer referral" do
+    let(:referral) { create(:referral, :complete) }
+
+    it "renders the name row" do
+      expect(page).to have_css("dd", text: referrer.name)
+      expect(page).to have_link "Change", href: "/referrals/#{referral.id}/referrer/name/edit?return_to=%2F"
+    end
```
[the remark this answers is not in the document]
Comment (paulodeon) [reply spec/components/about_you_component_spec.rb:22]: Added an ordering test

**Review by tavy88 — APPROVED** [review]
LGTM!

File: app/components/about_you_component.rb [hunk 3]
```diff
@@ -1,76 +1,40 @@
...
-            text: "Your job title"
-          },
-          value: {
-            text: nullable_value_to_s(referrer&.job_title)
-          }
-        }
-      )
-    end
+  def name_row
+    { label: "Your name", value: referrer&.name, path: [:edit, referral.routing_scope, referral, :referrer_name] }
```
Comment (gpeng) [inline app/components/about_you_component.rb:16]: Isn't the `path` bit always the same apart from the last argument?

Comment (paulodeon) [reply app/components/about_you_component.rb:16]: Added - I think it's look real tidy now

**Pull request merged**